src/Apps/Order/Routes/__tests__/Utils/OrderAppTestPage.tsx

```diff
@@ -0,0 +1,118 @@
+import { ShippingSummaryItemFragmentContainer } from "Apps/Order/Components/ShippingSummaryItem"
+import { TransactionDetailsSummaryItem } from "Apps/Order/Components/TransactionDetailsSummaryItem"
+import { ModalButton, ModalDialog } from "Components/Modal/ModalDialog"
+import { Stepper } from "Components/v2"
+import { CountdownTimer } from "Components/v2/CountdownTimer"
+import { expectOne, RootTestPage } from "DevTools/RootTestPage"
+
+export class OrderAppTestPage extends RootTestPage {
+  /** COMPONENT SELECTORS **/
+
+  get orderStepper() {
+    return expectOne(this.root.find(OrderStepper))
+  }
+
+  get orderStepperCurrentStep() {
+    const index = this.root.find(Stepper).props().currentStepIndex
+    return (this.root.find(OrderStepper).props() as any).steps[index]
+  }
+
+  get transactionSummary() {
+    return expectOne(this.root.find(TransactionDetailsSummaryItem))
+  }
+
+  get artworkSummary() {
+    return expectOne(this.root.find(ArtworkSummaryItemFragmentContainer))
+  }
+
+  get shippingSummary() {
+    return expectOne(this.root.find(ShippingSummaryItemFragmentContainer))
+  }
+
+  get paymentSummary() {
+    return expectOne(this.root.find(CreditCardSummaryItemFragmentContainer))
+  }
+
+  get countdownTimer() {
+    return expectOne(this.root.find(CountdownTimer))
+  }
+
+  get conditionsOfSaleDisclaimer() {
+    return expectOne(this.root.find(ConditionsOfSaleDisclaimer))
+  }
+
+  get modalDialog() {
+    return expectOne(this.root.find(ModalDialog))
+  }
+
+  get submitButton() {
+    return expectOne(this.find(Button).last())
+  }
+
+  get offerInput() {
+    return expectOne(this.find(OfferInput))
+  }
+
+  /** PAGE ACTIONS **/
+
+  async clickSubmit() {
+    this.submitButton.simulate("click")
+    await this.update()
+  }
+
+  async dismissModal() {
+    this.modalDialog
+      .find(ModalButton)
+      .last()
+      .simulate("click")
+    await this.update()
+  }
+
+  /*** COMMON ASSERTIONS ***/
+
+  async expectDefaultErrorDialog() {
+    await this.expectErrorDialogMatching(
+      "An error occurred",
+      "Something went wrong. Please try again or contact orders@artsy.net."
+    )
+  }
+
+  async expectErrorDialogMatching(
+    title: string,
+    message: string,
+    buttonText?: string
+  ) {
+    expect(this.modalDialog.props().show).toBe(true)
+    expect(this.modalDialog.text()).toContain(title)
+    expect(this.modalDialog.text()).toContain(message)
+    if (buttonText) {
+      const button = this.modalDialog.find(ModalButton)
+      expect(button.length).toBe(1)
+      expect(button.text()).toMatch(buttonText)
+    }
+
+    await this.dismissModal()
+
+    expect(this.modalDialog.props().show).toBe(false)
+  }
+
+  async setOfferAmount(amount: number) {
+    this.offerInput.props().onChange(amount)
+    await this.update()
+  }
+
+  async expectButtonSpinnerWhenSubmitting() {
+    expect(this.submitButton.props().loading).toBeFalsy()
+    this.clickSubmit()
+    this.root.update()
```

Comment:
yeah, `ReactWrapper.update` forces a re-render. It's needed after state changes that don't happen synchronously as a result of calling some other `ReactWrapper` method. e.g. when you do `wrapper.simulate('click')` the `onClick` event handler is called, and then afterwards `wrapper.update` is transparently invoked. If a call to `.setState` doesn't happen synchronously during that process, but asynchronously, then you need to call `.update` manually.

So coming back to this function, our `clickSubmit` synchronously changes the state of the 'loading' flag on the page, which should also call `.update` on the root wrapper, then we redundantly call `.update` again because I was being careless :DDD

Will remove, thanks! :pray:

Reply:
Have removed.